tests/workbenches/utils.py

```diff
@@ -1,11 +1,29 @@
 from kubernetes.dynamic import DynamicClient, Resource, ResourceInstance
+from kubernetes.dynamic.exceptions import ResourceNotFoundError
+from simple_logger.logger import get_logger
 
+LOGGER = get_logger(name="workbenches_utils")
 
-def get_username(dyn_client: DynamicClient) -> str:
+
+def get_username(dyn_client: DynamicClient) -> str | None:
     """Gets the username for the client (see kubectl -v8 auth whoami)"""
-    self_subject_review_resource: Resource = dyn_client.resources.get(
-        api_version="authentication.k8s.io/v1", kind="SelfSubjectReview"
-    )
-    self_subject_review: ResourceInstance = dyn_client.create(self_subject_review_resource)
-    username: str = self_subject_review.status.userInfo.username
+    username: str | None = None
+    try:
+        self_subject_review_resource: Resource = dyn_client.resources.get(
+            api_version="authentication.k8s.io/v1", kind="SelfSubjectReview"
+        )
+        self_subject_review: ResourceInstance = dyn_client.create(self_subject_review_resource)
+        username = self_subject_review.status.userInfo.username
+    except ResourceNotFoundError:
+        LOGGER.info(
+            "SelfSubjectReview not found. Falling back to user.openshift.io/v1/users/~ for OpenShift versions <=4.14"
+        )
+        user_response, status, _ = dyn_client.client.call_api(
```

Comment:
Please use https://github.com/RedHatQE/openshift-python-wrapper/pull/2387
When a resource is missing from the wrapper, we should add it there. There's a simple tool to add one resources (https://github.com/RedHatQE/openshift-python-wrapper/blob/main/class_generator/README.md)

This creates consistency within the project

Reply:
It may not be possible, because `~` is not a valid resource name in kubernetes/openshift. I'll try and report how it went.

Reply:
* https://github.com/RedHatQE/openshift-python-wrapper/pull/2387

Reply:
* https://github.com/RedHatQE/openshift-python-wrapper/pull/2389

Reply:
This PR should probably be marked as wip, till the wrapper changes are merged.